openmetadata-mcp/src/main/java/org/openmetadata/mcp/HttpServletSseServerTransportProvider.java

```diff
@@ -49,12 +49,7 @@ public class HttpServletSseServerTransportProvider extends HttpServlet
   private final AtomicBoolean isClosing = new AtomicBoolean(false);
   private McpServerSession.Factory sessionFactory;
   private ExecutorService executorService =
-      Executors.newCachedThreadPool(
-          r -> {
-            Thread t = new Thread(r, "MCP-Worker-SSE");
-            t.setDaemon(true);
-            return t;
-          });
+      Executors.newFixedThreadPool(10, Thread.ofVirtual().name("MCP-Worker-SSE", 0).factory());
```

Comment:
should we have the max threads as a config? In case we still drown instances and need to tune this

Reply:
It's to handle and open streams if the client supports it , process the request and close the stream
Since we moved to virtual threads now z blocking calls would not be an issue